warp10/src/main/java/io/warp10/script/functions/FETCH.java

```diff
@@ -572,16 +556,35 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
                   base.setMetadata(decoderMeta);
                   
                   // Force type attribute
-                  base.getMetadata().putToAttributes(typelabel, typename);
+                  base.getMetadata().putToAttributes(typeattr, typename);
                   if (null != uuid) {
                     base.getMetadata().putToAttributes(Constants.UUID_ATTRIBUTE, uuid.toString());
                   }
                 }
-                
-                GTSHelper.setValue(base, ts, location, elevation, value, false);                
+
+                //
+                // When using HBase, fetch requests which do not have boundaries may be served
+                // using a custom filter. This filter may return more data for a GTS than the requested
+                // count if the GTS spans multiple regions because the filtering is performed on the Region Servers
+                // and in this specific case the request will be forwarded to all RS serving regions for a given GTS.
+                // It is therefore necessary to keep track of how many datapoints were already fetched and
+                // shrink the GTS so we do not return too many.
+                //
+
+                if (countOnly) {
+                  if (lastCount + dpcount < count) {
+                    GTSHelper.setValue(base, ts, location, elevation, value, false);                
+                  } else {
+                    // We are done, exit
+                    break;
+                  }
+                } else {
+                  // Consider all points returned by the underlying system
+                  GTSHelper.setValue(base, ts, location, elevation, value, false);                
+                }
```

Comment:
Could be simplified by keeping `GTSHelper.setValue(base, ts, location, elevation, value, false);` here and adding 
```
if (countOnly && lastCount + dpcount >= count) {
  break;
}
```
earlier in the loop or even in the while condition.

Reply:
Keeping the conditional just above the call to setValue.